ci/pr-pipeline.yml

```diff
@@ -466,18 +493,6 @@ jobs:
                 OPERATOR_IMG_DIGEST=$(skopeo inspect --format "{{.Digest}}" docker://${OPERATOR_IMAGE_NAME}:${IMAGE_TAG})
                 echo "OPERATOR_IMG_DIGEST=$OPERATOR_IMG_DIGEST"
 
-                export PATH="$PATH:/usr/local/go/bin"
-                export GOPATH=/go
-                export GO_VERSION=1.22.3
-                export GO_SHA256="8920ea521bad8f6b7bc377b4824982e011c19af27df88a815e3586ea895f1b36 go${GO_VERSION}.linux-amd64.tar.gz"
-
-                curl -L --fail --show-error --silent "https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz" -o "go${GO_VERSION}.linux-amd64.tar.gz" \
-                && echo "${GO_SHA256}" | sha256sum --check \
-                && rm -rf /usr/local/go \
-                && tar -C /usr/local -xzf "go${GO_VERSION}.linux-amd64.tar.gz" \
-                && mkdir -p "${GOPATH}" \
-                && go version
-
```

Comment:
I am not fully sure, if we should still stick with explicitly installing golang versions in the build step.
If we ever run multiple PR pipelines concurrently, and the PRs have different golang versions, we might have timing side-effects, as one pipeline could install a different go version, pushes the base image and the other pipeline just picks the incorrect image.
Maybe we could extract the go install into a shell script and reuse it in all required locations so that we have the same go release inside the same pipeline run, but are isolated against concurrent pipeline runs?

Reply:
Each PR actually has its own base image, the testing should be able to done in isolated manner without any issues

Reply:
ah, okay, fair point, did not notice that.